### PR TITLE
added popreact interface and sheets api

### LINKED ISSUE #4: React JS setup for popup
Currently plain HML, CSS and JS is used for frontend. Need to migrate to React JS

### PR DESCRIPTION
**Pull request by manav1403**
fixes #4

### CODE REVIEW

File: react-setup/src/pages/Popup/popup.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+function AddAlarm_click(course_no,course_details, dates){
+  console.log(course_no, course_details,  dates);
+}
+
+export { AddAlarm_click }
```
Comment (hirenchalodiya1) [inline react-setup/src/pages/Popup/popup.js:5]: Why this file is created?

File: react-setup/src/pages/Popup/sheetapi.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      for(var key in formatteddata){
+        if(!formatteddata[key]['F'] || formatteddata[key]['F'] == 'slot') continue;
+        result[formatteddata[key]['F']] = '';
+      }
+      return Object.keys(result);
+    } else {
+      alert("HTTP-Error: " + response.status);
+    } 
+  }
+  get_slots();
```
Comment (hirenchalodiya1) [inline react-setup/src/pages/Popup/sheetapi.js:23]: is this function call necessary here?

**Review by hirenchalodiya1 — COMMENTED** [review]
Look at file comments

**Review by hirenchalodiya1 — APPROVED** [review]

File: react-setup/src/pages/Popup/popup.js (hunk 1 above)
Comment (manav1403) [reply react-setup/src/pages/Popup/popup.js:5]: this is to give a refrence of function to darshit to create further code in the file to add alarms

File: react-setup/src/pages/Popup/sheetapi.js (hunk 2 above)
Comment (manav1403) [reply react-setup/src/pages/Popup/sheetapi.js]: no, its not corrected in the latest push

File: react-setup/src/pages/Popup/popup.js (hunk 1 above)
Comment (hirenchalodiya1) [reply react-setup/src/pages/Popup/popup.js:5]: remove it from pr

Comment (manav1403) [reply react-setup/src/pages/Popup/popup.js]: removed!

**Review by hirenchalodiya1 — APPROVED** [review]

**Pull request merged**